src/devicecontroller/DefaultDeviceController.ts

```diff
@@ -492,21 +507,24 @@ export default class DefaultDeviceController implements DeviceControllerBasedMed
       return DevicePermission.PermissionGrantedByBrowser;
     }
 
-    const proposedConstraints: MediaStreamConstraints | null = this.calculateMediaStreamConstraints(
+    let proposedConstraints: MediaStreamConstraints | null = this.calculateMediaStreamConstraints(
       kind,
       device
     );
+
     if (
       this.activeDevices[kind] &&
       this.activeDevices[kind].matchesConstraints(proposedConstraints) &&
-      this.activeDevices[kind].stream.active
+      this.activeDevices[kind].stream.active &&
+      this.activeDevices[kind].groupId !== null &&
```

Comment:
This is not a correct logic since we want to skip groupId check if groupId is null. Also since this only happens to Chrome default, should we rework this to its own method and only check groupId for Chrome and default to reduce the change of bug?